### ISSUE TITLE
feature request: multi-publish

### ISSUE THREAD

**Issue by matthiasbeyer** [imp/cargo-multi#5]
Hi,

this looks like a great tool to do multi-crate publishing. What I mean: I have a project where a lot of crates live within the same repository ([imag](https://github.com/matthiasbeyer/imag)). All of these crates depend on each other (well, it is a tree-ish structure, not a circle of course). I released my first version (0.2.0) some days ago. I did it with this script:

``` bash
while true; do 
   for dir in *; pushd "$dir"; cargo publish; popd; done
done
```

(not _exactly_ that script, but you get the gist of it)

Another possibility would be to _manually_ find the dependency tree and do it by hand. _Or_, and that's what I'm suggesting: using a tool for it which internally gets the dependency tree of all crates to work on and `cargo publish` them in the right order.

TL;DR: allow me to `cargo multi publish` a set of crates that might depend on eachother and optimize this process to publish them in the right order (so, crates at the root of the dependency tree first, others later).

**Comment by imp** [comment]
Interesting idea. It means that `multi publish` will need to be special case. Nonetheless, it worth exploring
